### PR TITLE
Added project auto save 

### PR DESCRIPTION
**Pull request by igorkorsukov**
I do not know if something is needed more complicated, I did it simply, further experience will show whether it is necessary to complicate

### CODE REVIEW

File: src/project/internal/projectautosaver.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        return;
+    }
+
+    if (project->created().val) {
+        LOGD() << "[autosave] project just created";
+        return;
+    }
+
+    if (!project->needSave().val) {
+        LOGD() << "[autosave] project not need save";
```
Comment (Jojo-Schmitz) [inline src/project/internal/projectautosaver.cpp:51]: Typo: `does not` and probably `saving`

File: src/project/internal/projectautosaver.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return;
+    }
+
+    Ret ret = project->save();
+    if (!ret) {
+        LOGE() << "[autosave] failed save project, err: " << ret.toString();
```
Comment (Jojo-Schmitz) [inline src/project/internal/projectautosaver.cpp:57]: Typoe: `to save`

File: src/project/internal/projectautosaver.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return;
+    }
+
+    LOGD() << "[autosave] success project saved";
```
Comment (Jojo-Schmitz) [inline src/project/internal/projectautosaver.cpp:61]: Typoe: `successfully saved project` or `project saved successfully ` or `success saving project`

File: src/project/internal/projectautosaver.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/project/internal/projectautosaver.cpp]: Thanks, fixed!

File: src/project/internal/projectautosaver.cpp (hunk 2 above)
Comment (igorkorsukov) [reply src/project/internal/projectautosaver.cpp]: Thanks, fixed!

File: src/project/internal/projectautosaver.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/project/internal/projectautosaver.cpp]: Thanks, fixed!

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**